source/selection/tournament_selection.py

```diff
@@ -0,0 +1,24 @@
+import random
+from source.selection.selection_strategy import SelectionStrategy
+from source.population.individual import Individual
+
+
+class TournamentSelection(SelectionStrategy):
+    def select(self, individuals: list[Individual], fitness_function, selection_size: int, tournament_size: int = 3,
+               **kwargs) -> list[Individual]:
+        """
+        Selects individuals using tournament selection.
+
+        :param individuals: List of individuals in the population.
+        :param fitness_function: Fitness function to evaluate individuals.
+        :param selection_size: Number of individuals to select.
+        :param tournament_size: Number of individuals in each tournament.
+        :return: List of selected individuals.
+        """
+        selected_individuals = []
+        for _ in range(selection_size):
+            tournament = random.sample(individuals, tournament_size)
```

Comment:
to powinno być chyba losowanie bez powtórzeń tych osobników, więc na początku powinno się wybrać wszystkich ziomków i podzielić ich na turnieje

Reply:
Ok, racja. Pytanko: Co zrobić jeśli liczba osobników nie dzieli się bez reszty przez parametr tournament_size?

Reply:
to znaczy może się nie dzielić tylko selection_size *   tournament_size < liczba osobników 
Jeśli tak nie będzie to wyjątek jest okej

Reply:
Wtedy wszyscy osobnicy nie będą wybierani do losowania, i można stracić najlepszych, hmm... Dlatego może wywalę parametr selection_size, i stworzę tyle grup ile się zmieści?

Reply:
ja założyłam w krzyżowaniu, że ta lista będzie bez tych najlepszych osobników. Dlatego robię z nią co chce. Wydaje mi się ze to nie będzie cięzkie do ogarniecia w spięciu tego algorytmu

Reply:
Ok, przedyskutowane na callu, zamykam